types/mojang-net/index.d.ts

```diff
@@ -0,0 +1,172 @@
+// Project: https://docs.microsoft.com/minecraft/creator/
+// Definitions by: Jake Shirley <https://github.com/JakeShirley>
+//                 Mike Ammerlaan <https://github.com/mammerla>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/* *****************************************************************************
+   Copyright (c) Microsoft Corporation.
+   ***************************************************************************** */
+/**
+ * The `mojang-net` module contains types for executing
+ * HTTP-based requests.
+ *
+ * Manifest Details
+ * ```json
+ * {
+ *   // mojang-net
+ *   "uuid": "777b1798-13a6-401c-9cba-0cf17e31a81b",
+ *   "version": [ 0, 1, 0 ]
+ * }
+ * ```
+ *
+ */
+import * as mojangminecraftserveradmin from 'mojang-minecraft-server-admin';
+export enum HttpRequestMethod {
+    /**
+     * Represents the method for an HTTP PUT request. GET requests
+     * are commonly used to retrieve information about a resource
+     * at the specified URI.
+     */
+    POST = 0,
+    /**
+     * Represents the method for an HTTP PUT request. PUT requests
+     * are commonly used to update a single resource that already
+     * exists in a resource collection.
+     */
+    PUT = 1,
+    /**
+     * Represents the method for an HTTP PUT request. POST requests
+     * are commonly used to create a new resource that is a
+     * subordinate of the specified URI.
+     */
+    GET = 2,
+    /**
+     * Represents the method for an HTTP HEAD request. HEAD
+     * requests are similar to a GET request, but are commonly used
+     * to retrieve just the HTTP response headers from the
+     * specified URI, and not the body contents.
+     */
+    DELETE = 3,
+    /**
+     * Represents the method for an HTTP PUT request. GET requests
+     * are commonly used to retrieve information about a resource
+     * at the specified URI.
+     */
+    HEAD = 4,
+}
+export class HttpClient {
+    /**
+     * @remarks
+     * Cancels all pending requests.
+     * @param reason
+     */
+    cancelAll(reason: string): void;
+    /**
+     * @remarks
+     * Performs a simple HTTP get request.
+     * @param uri
+     * URL to make an HTTP Request to.
+     * @returns
+     * An awaitable promise that contains the HTTP response.
+     */
+    get(uri: string): Promise<HttpResponse>;
+    /**
+     * @remarks
+     * Performs an HTTP request.
+     * @param config
+     * Contains an HTTP Request object with configuration data on
+     * the HTTP request.
+     * @returns
+     * An awaitable promise that contains the HTTP response.
+     */
+    request(config: HttpRequest): Promise<HttpResponse>;
+    protected constructor();
+}
+/**
+ * Represents an HTTP header - a key/value pair of
+ * meta-information about a request.
+ */
+export class HttpHeader {
+    /**
+     * Key of the HTTP header.
+     */
+    'key': string;
+    /**
+     * Value of the HTTP header.
+     */
+    'value': string;
```

Comment:
The constructor's `value` is a variant between the secret string and a regular string. How does this behave when the stored value is a secret string? Is a null returned? An empty string?

Reply:
We'll fix this with a change to the signature in future updates.